src/textual/app.py

```diff
@@ -922,6 +928,59 @@ def active_bindings(self) -> dict[str, ActiveBinding]:
         """
         return self.screen.active_bindings
 
+    def get_system_commands(self, screen: Screen) -> SystemCommandsResult:
+        """A generator of system commands used in the command palette.
+
+        Args:
+            screen: The screen where the command palette was invoked from.
+
+        Implement this method in your App subclass if you want to add custom commands.
+        Here is an example:
+
+        ```python
+        def get_system_commands(self):
+            yield from super().get_system_commands()
+            yield ("Bell", "Ring the bell", self.bell)
+        ```
+
+        !!! note
+            Requires that [`SystemCommandsProvider`][textual.system_commands.SystemCommandsProvider] is in `App.COMMANDS` class variable.
+
+        Yields:
+            tuples of (TITLE, HELP TEXT, CALLBACK)
```

Comment:
IMO a NamedTuple or something would be cleaner here as you'd be able to name and document each field. I always use the `Binding` class instead of tuples for example because I cannot remember the order and would have to check the docs each time. Using `Binding` makes it easy to remember because my IDE helps.

An optional 4th bool which determines whether to include/exclude commands from the "discovery" section without needing to create a custom provider would also be super useful.

Reply:
Ok, done.

Personally I find atomic types and a docstring to be less taxing. But a NT is more IDE friendly.

Reply:
Thank you 🙏